robolectric/src/main/java/org/robolectric/util/ActivityController.java

```diff
@@ -241,4 +242,71 @@ public void run() {
         .resume()
         .visible();
   }
+  
+  /**
+   * Performs a configuration change on the Activity.
+   *  
+   * @param newConfiguration The new configuration to be set.
+   * @return May return a new ActivityController for a new Activity instance if the configuration change
+   * 		is not handled by the activity itself, otherwise, itself.
+   */
+  public ActivityController<T> configurationChange(final Configuration newConfiguration) {
+    final Configuration currentConfig = component.getResources().getConfiguration();
+    final int changedBits = currentConfig.diff(newConfiguration);
+    currentConfig.setTo(newConfiguration);
+    
+    // Can the activity handle itself ALL configuration changes?
+    if ((getActivityInfo(component.getApplication()).configChanges & changedBits) == changedBits) {
+      shadowMainLooper.runPaused(new Runnable() {
+        @Override
+        public void run() {
+          ReflectionHelpers.callInstanceMethod(Activity.class, component, "onConfigurationChanged",
+            ClassParameter.from(Configuration.class, newConfiguration));
+        }
+      });
+
+      return this;
+    } else {
+    	@SuppressWarnings("unchecked")
+		final ActivityController<T> recreatedActivityController = Robolectric.buildActivity(
```

Comment:
No, I meant updating the component field. Its OK to make it non-final because it can obviously change according to the Android semantics. This method should still return "this" builder, which represents the Android environment which doesn't change.